sparse_strips/vello_common/src/recording.rs

```diff
@@ -0,0 +1,410 @@
+// Copyright 2025 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Recording API for caching sparse strips
+
+use crate::kurbo::{Affine, BezPath, Rect, Stroke};
+use crate::mask::Mask;
+use crate::paint::PaintType;
+use crate::peniko::{BlendMode, Fill};
+use crate::strip::Strip;
+use alloc::boxed::Box;
+use alloc::vec::Vec;
+
+/// Cached sparse strip data
+#[derive(Debug, Clone)]
+pub struct CachedStrips {
+    /// The cached sparse strips
+    pub strips: Box<[Strip]>,
+    /// The alpha buffer data
+    pub alphas: Box<[u8]>,
+}
+
+impl CachedStrips {
+    /// Create a new cached strips instance
+    pub fn new(strips: Box<[Strip]>, alphas: Box<[u8]>) -> Self {
+        Self { strips, alphas }
+    }
+
+    /// Check if this cached strips is empty
+    pub fn is_empty(&self) -> bool {
+        self.strips.is_empty()
+    }
+
+    /// Get the number of strips
+    pub fn strip_count(&self) -> usize {
+        self.strips.len()
+    }
+
+    /// Get the number of alpha bytes
+    pub fn alpha_count(&self) -> usize {
+        self.alphas.len()
+    }
+
+    /// Get strips as slice
+    pub fn strips(&self) -> &[Strip] {
+        &self.strips
+    }
+
+    /// Get alphas as slice
+    pub fn alphas(&self) -> &[u8] {
+        &self.alphas
+    }
+}
+
+/// A recording of rendering commands that can cache generated strips
+#[derive(Debug, Clone)]
+pub struct Recording {
```

Comment:
Do we have a strategy so that consumers of `vello_cpu` and `vello_hybrid` cannot access internal-to-vello fields? Since `Recording`'s fields are public, we can access them entirely by consumers (this is a way for consumers to break invariants).

For example, a silly consumer can do this:

<img width="1054" height="477" alt="image" src="https://github.com/user-attachments/assets/dee82a1c-a5b1-40e0-9446-5b4237eee79a" />


---

Don't need to solve this here - I'm just wondering whether it's something we've already thought about.